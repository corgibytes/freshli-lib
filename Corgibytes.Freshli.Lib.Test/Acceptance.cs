using System;
using ApprovalTests;
using Xunit;

namespace Corgibytes.Freshli.Lib.Test
{
    public class Acceptance
    {

        private DateTimeOffset _testingBoundary =
          new DateTimeOffset(2020, 02, 01, 0, 0, 0, 0, TimeSpan.Zero);

        [Fact]
        public void RubyGemsWithGitHistory()
        {
            var runner = new Runner();

            var rubyFixturePath = Fixtures.Path("ruby", "nokotest");
            var results = runner.Run(rubyFixturePath, asOf: _testingBoundary);

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void RubyGemsWithHistoryViaGitHub()
        {
            var runner = new Runner();

            var repoUrl =
              "https://github.com/corgibytes/freshli-fixture-ruby-nokotest";
            var results = runner.Run(repoUrl, asOf: _testingBoundary);

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void RubyGemsFeedbinHistoryViaGitHub()
        {
            var runner = new Runner();

            var repoUrl = "https://github.com/feedbin/feedbin";
            var results = runner.Run(repoUrl, asOf: _testingBoundary);

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void RubyGemsClearanceHistoryViaGitHub()
        {
            var runner = new Runner();
            var results = runner.Run(
              "https://github.com/thoughtbot/clearance",
              asOf: new DateTimeOffset(2020, 06, 01, 00, 00, 00, 00, TimeSpan.Zero)
            );

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void ComposerWithoutGitHistory()
        {
            var runner = new Runner();

            var phpFixturePath = Fixtures.Path("php", "large");
            var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void DrupalComposerWithoutGitHistory()
        {
            var runner = new Runner();

            var phpFixturePath = Fixtures.Path("php", "drupal");
            var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void RequirementsTxtPyspider()
        {
            var runner = new Runner();

            var results = runner.Run(
              "https://github.com/binux/pyspider",
              asOf: _testingBoundary
            );

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void CpanfileDancer2()
        {
            var runner = new Runner();

            var results = runner.Run(
              "https://github.com/PerlDancer/Dancer2",
              asOf: _testingBoundary
            );

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void SpaCyWithHistoryViaGitHub()
        {
            var runner = new Runner();

            var results = runner.Run(
              "https://github.com/explosion/spaCy",
              asOf: new DateTimeOffset(2017, 06, 01, 00, 00, 00, TimeSpan.Zero)
            );

            Assert.True(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }

        [Fact]
        public void UnsupportedGitRepository()
        {
            var runner = new Runner();

            var results = runner.Run(
              "https://github.com/gohugoio/hugo",
              asOf: _testingBoundary
            );

            Assert.False(runner.ManifestFinder.Successful);
            Approvals.VerifyAll(results, "results");
        }
    }
}
