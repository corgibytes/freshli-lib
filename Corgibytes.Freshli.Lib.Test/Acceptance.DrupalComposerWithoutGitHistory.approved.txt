results[0] = Filename: composer.lock
Results:
{ Date: 2020-01-01T00:00:00, ManifestSHA: N/A, LibYear: { _packagesValues: [
{ Name: "drupal/composer_autoloader", RepoVersion: "1.0.0", RepoVersionPublishedAt: 2014-12-24T00:00:00, LatestVersion: "1.3.0", LatestPublishedAt: 2018-03-11T00:00:00, UpgradeAvailable: True, Value: 3.213698630136986 },
{ Name: "drupal/drupal", RepoVersion: "7.69.0", RepoVersionPublishedAt: 2020-05-20T00:00:00, LatestVersion: "8.8.1", LatestPublishedAt: 2019-12-18T00:00:00, UpgradeAvailable: True, Value: 0 },
], Total: 3.213698630136986 } }

